compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmInstructionGen.java

```diff
@@ -1583,14 +1631,17 @@ void generateFPLoadIns(BIRNonTerminator.FPLoad inst, LambdaMetadata lambdaMetada
 
     void generateNewXMLElementIns(BIRNonTerminator.NewXMLElement newXMLElement) {
 
-        this.loadVar(newXMLElement.startTagOp.variableDcl);
-        this.mv.visitTypeInsn(CHECKCAST, XML_QNAME);
-        this.loadVar(newXMLElement.defaultNsURIOp.variableDcl);
-        this.mv.visitMethodInsn(INVOKESTATIC, XML_FACTORY, "createXMLElement",
-                String.format("(L%s;L%s;)L%s;", XML_QNAME,
-                        JvmPackageGen.IS_BSTRING ? JvmConstants.B_STRING_VALUE : STRING_VALUE, XML_VALUE), false);
-        this.storeToVar(newXMLElement.lhsOp.variableDcl);
-    }
+            this.loadVar(newXMLElement.startTagOp.variableDcl);
+            this.mv.visitTypeInsn(CHECKCAST, XML_QNAME);
+            this.loadVar(newXMLElement.defaultNsURIOp.variableDcl);
+            loadType(this.mv, newXMLElement.type);
+
+            this.mv.visitMethodInsn(INVOKESTATIC, XML_FACTORY, "createXMLElement",
+                                    String.format("(L%s;L%s;L%s;)L%s;", XML_QNAME,
+                                                  JvmPackageGen.IS_BSTRING ? JvmConstants.B_STRING_VALUE :
+                                                          STRING_VALUE, BTYPE, XML_VALUE), false);
+            this.storeToVar(newXMLElement.lhsOp.variableDcl);
+        }
```

Comment:
looks like there is an alignment issue here. probably due to merge conflicts with jvm codegen package classes.